pyproject.toml

```
@@ -0,0 +1,150 @@
+[build-system]
+requires = ["setuptools>=61.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "mhkit"
+# `version` is read from `[tools.setuptools.dynamic] version` during build: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata
+dynamic = ["version"]
+description = "Marine and Hydrokinetic Toolkit"
+readme = "README.md"
+authors = [
+    {name = "MHKiT developers"}
+]
+license = {text = "Revised BSD"}
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Programming Language :: Python :: 3",
+    "Topic :: Scientific/Engineering",
+    "Intended Audience :: Science/Research",
+    "Operating System :: OS Independent",
+]
+requires-python = ">=3.10"
+dependencies = [
+    "numpy>=2.0.0",
+    "pandas>=2.2.2",
+    "scipy>=1.14.0",
+    "xarray>=2024.6.0",
+    "matplotlib>=3.9.1",
+]
+
+[project.optional-dependencies]
+# Core dependencies for each module
+wave = [
+    "scikit-learn>=1.5.1",
+    "statsmodels>=0.14.2",
+    "numexpr>=2.10.0",
+    "bottleneck",
+]
+
+tidal = [
+    "scikit-learn>=1.5.1",
+    "statsmodels>=0.14.2",
+    "numexpr>=2.10.0",
+    "bottleneck",
+]
+
+river = [
+    "scikit-learn>=1.5.1",
+    "statsmodels>=0.14.2",
+    "numexpr>=2.10.0",
+    "bottleneck",
+]
+
+dolfyn = [
+    "h5py>=3.11.0",
+    "h5pyd>=0.18.0",
+    "netCDF4>=1.7.1.post1",
+    "cartopy",
+]
+
+power = [
+    "scikit-learn>=1.5.1",
+    "statsmodels>=0.14.2",
+]
+
+loads = [
+    "scikit-learn>=1.5.1",
+    "statsmodels>=0.14.2",
+]
+
+mooring = [
+    "scikit-learn>=1.5.1",
+    "statsmodels>=0.14.2",
+]
+
+acoustics = [
+    "scikit-learn>=1.5.1",
+    "statsmodels>=0.14.2",
+]
+
+qc = [
+    "scikit-learn>=1.5.1",
+    "statsmodels>=0.14.2",
+]
+
+utils = [
+    "requests",
+    "beautifulsoup4",
+    "lxml",
+    "pytz",
+    "NREL-rex>=0.2.63",
+    "pecos>=0.3.0",
+    "fatpack",
+]
+
+# Development dependencies
+dev = [
+    "pytest",
+    "pylint",
+    "pytest-cov",
+    "pre-commit",
+    "coverage",
+    "coveralls"
+]
+
+# Examples dependencies
+examples = [
+    "jupyter",
+    "notebook",
+    "ipykernel",
+    "nbval",
+    "utm",
+    "folium",
+    "mhkit[wave]",
+    "mhkit[tidal]",
+    "mhkit[river]",
+    "mhkit[dolfyn]",
+    "mhkit[power]",
+    "mhkit[loads]",
+    "mhkit[mooring]",
+    "mhkit[acoustics]",
+    "mhkit[qc]",
+    "mhkit[utils]",
```

simplify using "all" to cover all source code dependencies
```suggestion
    "mhkit[all]",